### PR TITLE
[OPIK-486] Add NOT_EQUAL operator for filtering

### PR DESCRIPTION
**Pull request by BorisTkachenko**
## Details
Add NOT_EQUAL operator for filtering

Resolves: #551

## Testing
Added corresponding integration tests

### CODE REVIEW

File: apps/opik-backend/src/main/java/com/comet/opik/api/filter/Operator.java [hunk 1]
```diff
@@ -12,6 +12,7 @@ public enum Operator {
     STARTS_WITH("starts_with"),
     ENDS_WITH("ends_with"),
     EQUAL("="),
+    NOT_EQUAL("!="),
```
Comment (andrescrz) [inline apps/opik-backend/src/main/java/com/comet/opik/api/filter/Operator.java:15]: Just make sure FrontEnd is happy with this syntax `!=` for the query param. From my side, it looks good.

File: apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/SpansResourceTest.java [hunk 2]
```diff
@@ -1109,15 +1110,56 @@ void getByProjectName__whenFilterByCorrespondingField__thenReturnSpansFiltered(S
                     apiKey);
         }
 
+        @ParameterizedTest
+        @MethodSource("equalAndNotEqualFilters")
+        void getByProjectName__whenFilterTotalEstimatedCostEqual_NotEqual__thenReturnSpansFiltered(Operator operator,
```
Comment (andrescrz) [inline apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/SpansResourceTest.java:1115]: There's similar tests for equals in traces object (and maybe some other). It'd be nice to add a test for all where this applies, not only spans.

**Review by andrescrz — APPROVED** [review]
A suggestion for tests, but not a blocker. You can add it in a follow-up PR.

File: apps/opik-backend/src/main/java/com/comet/opik/api/filter/Operator.java (hunk 1 above)
Comment (BorisTkachenko) [reply apps/opik-backend/src/main/java/com/comet/opik/api/filter/Operator.java:15]: FE confirmed that they are good: https://comet-ml.slack.com/archives/C0768KH9MDJ/p1733391186958639

File: apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/SpansResourceTest.java (hunk 2 above)
Comment (BorisTkachenko) [reply apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/SpansResourceTest.java:1115]: Will add in a later PR, we also have filters in Traces and Datasets.

**Pull request merged**